src/content/en/fundamentals/performance/webpack/decrease-frontend-size.md

```diff
@@ -0,0 +1,691 @@
+      'definitions. Use a static property named `defaultProps` instead.'
+    );
+    // …
+
+Such checks and warnings are usually unnecessary in production, but they remain in the code and
+increase the library size. Configure webpack to remove them with the
+[`DefinePlugin`](https://webpack.js.org/plugins/define-plugin/):
+
+     // webpack.config.js
+    const webpack = require('webpack');
+
+    module.exports = {
+      plugins: [
+        new webpack.DefinePlugin({
+          'process.env.NODE_ENV': '"production"',
+        }),
+        new webpack.optimize.UglifyJsPlugin(),
+      ],
+    };
+
+The `DefinePlugin` replaces all occurrences of a specified variable with a specific value. With the
+config from above:
+
+<ol>
+<li>
+
+The <code>DefinePlugin</code> will replace all occurrences of <code>process.env.NODE_ENV</code> with
+<code>"production"</code>:
+
+<pre class="prettyprint">
+ // vue/dist/vue.runtime.esm.js
+if (typeof val === 'string') {
+  name = camelize(val);
+  res[name] = { type: null };
+} else if (<strong>process.env.NODE_ENV !== 'production'</strong>) {
+  warn('props must be strings when using array syntax.');
+}
+</pre>
+
+↓
+
+<pre class="prettyprint">
+// vue/dist/vue.runtime.esm.js
+if (typeof val === 'string') {
+  name = camelize(val);
+  res[name] = { type: null };
+} else if (<strong>"production" !== 'production'</strong>) {
+  warn('props must be strings when using array syntax.');
+}
+</pre>
+
+</li>
+</ol>
+
+Note: If you prefer to configure environment variables via CLI, take a look at the
+[EnvironmentPlugin](https://webpack.js.org/plugins/environment-plugin/). It works like the
+`DefinePlugin`, but reads the environment and replaces `process.env.` expressions automatically.
+
+<ol start="2">
+<li>
+
+And then the <code>UglifyJsPlugin</code> will remove all such <code>if</code> branches – because
+<code>"production" !== 'production'</code> is always false, and the plugin understands that the code
+inside these branches will never execute:
+
+<pre class="prettyprint">
+// vue/dist/vue.runtime.esm.js
+if (typeof val === 'string') {
+  name = camelize(val);
+  res[name] = { type: null };
+} else if ("production" !== 'production') {
+  warn('props must be strings when using array syntax.');
+}
+</pre>
+
+↓
+
+<pre class="prettyprint">
+// vue/dist/vue.runtime.esm.js (without minification)
+if (typeof val === 'string') {
+  name = camelize(val);
+  res[name] = { type: null };
+}
+</pre>
+
+</li>
+</ol>
+
+Note: You’re not required to use the `UglifyJsPlugin`. You can use any different minifier as soon as
+it supports dead code removal (e.g., the [Babel Minify
+plugin](https://github.com/webpack-contrib/babel-minify-webpack-plugin) or the [Google Closure
+Compiler plugin](https://github.com/roman01la/webpack-closure-compiler)).
+
+### Further reading {: .hide-from-toc }
+
+* [What “environment variables” are](https://superuser.com/questions/284342/
```

Comment:
don't linebreak URLs, otherwise the link breaks. it's OK if this exceeds 80 chars